hivemind/p2p/p2p_daemon.py

```diff
@@ -0,0 +1,45 @@
+import subprocess
+import typing as tp
+
+
+class P2P(object):
+    """
+    Forks a child process and executes p2pd command with given arguments.
+    Sends SIGKILL to the child in destructor and on exit from contextmanager.
+    """
+
+    LIBP2P_CMD = 'p2pd'
+
+    def __init__(self, *args, **kwargs):
+        self._child = subprocess.Popen(args=self._make_process_args(args, kwargs))
+        try:
+            stdout, stderr = self._child.communicate(timeout=0.2)
+        except subprocess.TimeoutExpired:
+            pass
+        else:
+            raise RuntimeError(f'p2p daemon exited with stderr: {stderr}')
+
+    def __enter__(self):
+        return self._child
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self._kill_child()
+
+    def __del__(self):
+        self._kill_child()
```

Comment:
[Nitpicking] This will handle del/dereferencing, but not sigkill. Is there perhaps a way to create a non-unix-daemon (aka python daemon) that is guaranteed to die with the host process?